src/System.Private.ServiceModel/src/System/ServiceModel/Security/ReceiveSecurityHeader.cs

```diff
@@ -31,9 +57,589 @@ public override string Namespace
             get { return this.StandardsManager.SecurityVersion.HeaderNamespace.Value; }
         }
 
+        public Collection<SecurityToken> BasicSupportingTokens
+        {
+            get
+            {
+                return _basicTokens;
+            }
+        }
+
+        public Collection<SecurityToken> SignedSupportingTokens
+        {
+            get
+            {
+                return _signedTokens;
+            }
+        }
+
+        public Collection<SecurityToken> EndorsingSupportingTokens
+        {
+            get
+            {
+                return _endorsingTokens;
+            }
+        }
+
+        public Collection<SecurityToken> SignedEndorsingSupportingTokens
+        {
+            get
+            {
+                return _signedEndorsingTokens;
+            }
+        }
+
+        public Message ProcessedMessage
+        {
+            get { return Message; }
+        }
+
+        public SecurityTokenAuthenticator DerivedTokenAuthenticator
+        {
+            get
+            {
+                return _derivedTokenAuthenticator;
+            }
+            set
+            {
+                ThrowIfProcessingStarted();
+                _derivedTokenAuthenticator = value;
+            }
+        }
+
+        public bool EnforceDerivedKeyRequirement
+        {
+            get
+            {
+                return _enforceDerivedKeyRequirement;
+            }
+            set
+            {
+                ThrowIfProcessingStarted();
+                _enforceDerivedKeyRequirement = value;
+            }
+        }
+
+        public bool ReplayDetectionEnabled
+        {
+            get { return _replayDetectionEnabled; }
+            set
+            {
+                ThrowIfProcessingStarted();
+                _replayDetectionEnabled = value;
+            }
+        }
+
         protected override void OnWriteHeaderContents(XmlDictionaryWriter writer, MessageVersion messageVersion)
         {
             throw ExceptionHelper.PlatformNotSupported();
         }
```

Comment:
Does this PNSE have an implementation that we should be putting in (and perhaps commenting out with an Issue31 comment)?